### PR TITLE
[CARBONDATA-3952] After reset query not hitting MV

### PR DESCRIPTION
**Pull request by ShreelekhyaG**
### Why is this PR needed?
 After reset query not hitting MV.
 With the `reset`, `spark.sql.warehouse.dir` and `carbonStorePath` don't match and the databaseLocation will change to old table path format. So, new tables that are created after reset, take a different path incase of default.

Closing this PR, as it is identified as spark bug. More details can be found at https://issues.apache.org/jira/browse/SPARK-31234

 ### What changes were proposed in this PR?
1. Made changes to get the previously configured spark directory. (value of `spark.sql.warehouse.dir` before reset). now newly created tables take same path as before.
2.  After reset, if we create a table and mv, they both are taking different paths. for table creation, it is appended with hdfs://  after getting the database location and is stored in hdfs whereas the mv table is not appended so it's stored in the local path. To maintain the same store location, added `checkAndAppendFileSystemURIScheme` to the database location.

 ### Does this PR introduce any user interface change?
 - No

 ### Is any new testcase added?
 - Yes

### CODE REVIEW

**Comment by CarbonDataQA1** [comment]
Build Failed  with Spark 2.4.5, Please check CI http://121.244.95.60:12545/job/ApacheCarbon_PR_Builder_2.4.5/1973/

**Comment by CarbonDataQA1** [comment]
Build Failed  with Spark 2.3.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/3712/

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.3.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/3730/

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.4.5, Please check CI http://121.244.95.60:12545/job/ApacheCarbon_PR_Builder_2.4.5/1991/

File: integration/spark-common-cluster-test/src/test/scala/org/apache/carbondata/cluster/sdv/generated/SetParameterTestCase.scala [hunk 1]
```diff
@@ -252,6 +252,27 @@ class SetParameterTestCase extends QueryTest with BeforeAndAfterAll {
...
+    sql("CREATE TABLE maintable(empno int,empname string,projectcode int, projectjoindate " +
+        "Timestamp, projectenddate date,salary double) STORED AS carbondata")
+    sql("CREATE MATERIALIZED VIEW mv1 as select timeseries(projectenddate,'day'), sum" +
+        "(projectcode) from maintable group by timeseries(projectenddate,'day')")
+    sql("insert into maintable select 1000,'PURUJIT',00012,'2015-07-26 12:07:28','2016-05-20'," +
+        "15000.00")
+    sql("insert into maintable select 1001,'PANKAJ',00010,'2015-07-26 17:32:20','2016-05-20'," +
+        "25000.00")
+    sql("set carbon.input.segments.defualt.maintable=1")
+    checkExistence(sql("EXPLAIN select timeseries(projectenddate,'day'), sum(projectcode) from " +
```
Comment (Indhumathi27) [inline integration/spark-common-cluster-test/src/test/scala/org/apache/carbondata/cluster/sdv/generated/SetParameterTestCase.scala:267]: Can use 'TestUtil.verifyMVHit' method to verify if query hits mv or not

File: integration/spark-common-cluster-test/src/test/scala/org/apache/carbondata/cluster/sdv/generated/SetParameterTestCase.scala [hunk 2]
```diff
@@ -252,6 +252,27 @@ class SetParameterTestCase extends QueryTest with BeforeAndAfterAll {
...
+                       "maintable group by timeseries(projectenddate,'day')"), true, "mv1")
+    sql("reset")
+    checkExistence(sql("EXPLAIN select timeseries(projectenddate,'day'), sum(projectcode) from " +
```
Comment (Indhumathi27) [inline integration/spark-common-cluster-test/src/test/scala/org/apache/carbondata/cluster/sdv/generated/SetParameterTestCase.scala:270]: same as above comment

File: integration/spark-common-cluster-test/src/test/scala/org/apache/carbondata/cluster/sdv/generated/SetParameterTestCase.scala (hunk 1 above)
Comment (ShreelekhyaG) [reply integration/spark-common-cluster-test/src/test/scala/org/apache/carbondata/cluster/sdv/generated/SetParameterTestCase.scala]: ok added check with `verifyMVHit`.

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.3.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/3885/

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.4.5, Please check CI http://121.244.95.60:12545/job/ApacheCarbon_PR_Builder_2.4.5/2144/

**Comment by akashrn5** [comment]
@ShreelekhyaG i feel, instead of fixing the impacted MV issue, we need to check why reset changes the warehouse and carbon store, because now if we just fix MV, may be later we may get the issue with other features. Please check once.

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.3.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/3932/

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.4.5, Please check CI http://121.244.95.60:12545/job/ApacheCarbon_PR_Builder_2.4.5/2191/

**Comment by akashrn5** [comment]
@ShreelekhyaG as discussed this is fixed in spark, as it was a spark bug, so can you please close this PR. Please provide the spark jira link in both PR and CARBONDATA-3952  jira and close both PR and jira

**Comment by ShreelekhyaG** [comment]
Closing this PR, as it is identified as spark bug. More details can be found at https://issues.apache.org/jira/browse/SPARK-31234

**Pull request closed without merging**